### PR TITLE
Add Sphinx docs

### PR DESCRIPTION
**Pull request by cgmb**
I tried to make it somewhat match rocThrust.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+#
+# source_suffix = ['.rst', '.md']
+source_suffix = '.rst'
+
+# The master toctree document.
+master_doc = 'index'
+
+# General information about the project.
+project = u'hipCUB'
+copyright = u'2010-2011, Duane Merrill; 2011-2018, NVIDIA CORPORATION; Modifications Copyright 2019 Advanced Micro Devices'
```
Comment (saadrahim) [inline docs/conf.py:56]: I would check this line

Comment (cgmb) [reply docs/conf.py:56]: That's copied from [LICENSE.txt](https://github.com/ROCmSoftwarePlatform/hipCUB/blob/1bc92d7af3e7846cfa55ed82af6fb52ac8a4122c/LICENSE.txt)

Comment (saadrahim) [reply docs/conf.py:56]: Copyright 2019-2021 Advanced Micro Devices
If you have a moment, the license file should be updated as well.

**Comment by cgmb** [comment]
A quick link to the docs on Jenkins: http://math-ci.rocm.amd.com/job/compute-rocm-dkms-no-npi-hipclang/job/Static%20Analysis/job/hipCUB/job/PR-126/Documentation/

**Pull request merged**